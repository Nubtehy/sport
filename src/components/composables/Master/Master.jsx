import React from 'react';
import PropTypes from 'prop-types';

import Header from 'components/composables/Header';

import Footer from 'components/composables/Footer';


const displayName = 'Master';

const propTypes = {
  children: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};

const defaultProps = {

};

function Master({
  children,
  title,
}) {
  return (
    <div>
      <Header title={title}
      />
      <main>
        {children}
      </main>
      <Footer />
    </div>
  );
}

Master.displayName = displayName;
Master.propTypes = propTypes;
Master.defaultProps = defaultProps;

export default Master;
