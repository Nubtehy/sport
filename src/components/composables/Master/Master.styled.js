import styled from 'styled-components';

export const CartContainer = styled.div`
  width: 1170px;
  max-width: 100%;
  margin: 0 auto;
`;

export const CartMainContainer = styled.div`
  flex: 1 1 auto;
`;
export const MasterWrapper = styled.div`
  height: 100%;
  display:flex;
  flex-direction: column;
`;
