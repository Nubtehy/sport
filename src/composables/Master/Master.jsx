import React from 'react';
import PropTypes from 'prop-types';

import Header from 'composables/Header';

import Footer from 'composables/Footer';


const displayName = 'Master';

const propTypes = {
  children: PropTypes.string.isRequired,
};

const defaultProps = {

};

function Master({
  children
}) {
  return (
    <div>
      <Header />
      <main>
        {children}
      </main>
      <Footer />
    </div>
  );
}

Master.displayName = displayName;
Master.propTypes = propTypes;
Master.defaultProps = defaultProps;

export default Master;
