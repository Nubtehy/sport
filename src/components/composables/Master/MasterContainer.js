import {
  compose, withState, withHandlers, setDisplayName, mapProps,
} from 'recompose';
import { connect } from 'react-redux';
import { addToCart, removeProduct, fetchProducts } from 'actions';
import { getCartProducts, getTotal } from 'selectors';

import Master from './Master';

export const enhance = compose(
  setDisplayName('AboutContainer'),
  connect(state => ({
    count: getCartProducts(state).length,
  })),
  withState('name', 'setName', ''),
  mapProps(props => ({
    ...props,
    count,
  })),
);

export default enhance(Master);
