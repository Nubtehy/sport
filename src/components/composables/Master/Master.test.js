import React from 'react';
import { shallow } from 'enzyme';

import Master from './Master';

const testProps ={ children: '', total: 12, title: 'test' }


describe('Given a Master component', () => {
	describe('when the component is rendered', () => {
		let component;

		beforeEach(() => {
			component = shallow(
				<Master { ...testProps }>
					<div>Page</div>
				</Master>
			);
		});

		it('should match the snapshot', () => {
			expect(component).toMatchSnapshot();
		});
	});
});
