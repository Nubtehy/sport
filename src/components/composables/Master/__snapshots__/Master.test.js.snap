// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Master component when the component is rendered should match the snapshot 1`] = `
<Masterstyled__MasterWrapper>
  <Header
    title="test"
    total={12}
  />
  <Masterstyled__CartMainContainer>
    <Masterstyled__CartContainer>
      <div>
        Page
      </div>
    </Masterstyled__CartContainer>
  </Masterstyled__CartMainContainer>
  <Footer />
</Masterstyled__MasterWrapper>
`;
